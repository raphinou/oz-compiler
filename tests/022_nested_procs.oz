% P uses a global variable after a child proc (T) definition used it.
% In this case, the new local for P was triggered by a child proc definition
% before a direct use of the variable by the procedure.
% We need to check that the symbol for A in the call {T A} is the one
% referenced by the symbol for A in the definition of T
% The inner proc also uses the outer proc's argument directly: the outer proc's
% argument is a global to the inner proc
local
   A P B
in
   proc {P V}
      T
   in
      proc {T U}
        A=44
      in
        {Show A}
        {Show U}
        {Show V}
      end
      {T A} % use it after a nested proc requested a new local for it
   end
   A = 5
   B = 7
   {P A}
   {For 1 B 1 P}
end
