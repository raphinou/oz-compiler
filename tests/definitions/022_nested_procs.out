44
5
5
44
5
1
44
5
2
44
5
3
44
5
4
44
5
5
44
5
6
44
5
7
